homeassistant/components/sensor/co2signal.py

```diff
@@ -0,0 +1,109 @@
+"""
+Support for the CO2signal platform.
+For more details about this platform, please refer to the documentation
+"""
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+
+from homeassistant.const import (
+    CONF_TOKEN, CONF_LATITUDE, CONF_LONGITUDE)
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.helpers.entity import Entity
+
+CONF_COUNTRY_CODE = "country_code"
+
+REQUIREMENTS = ['co2signal==0.4.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTRIBUTION = 'Data provided by CO2signal'
+
+MSG_LOCATION = "Please use either coordinates or the country code. " \
+               "For the coordinates, " \
+               "you need to use both latitude and longitude."
+CO2_INTENSITY_UNIT = "CO2eq/kWh"
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_TOKEN): cv.string,
+    vol.Inclusive(CONF_LATITUDE, 'coords', msg=MSG_LOCATION): cv.latitude,
+    vol.Inclusive(CONF_LONGITUDE, 'coords', msg=MSG_LOCATION): cv.longitude,
+    vol.Optional(CONF_COUNTRY_CODE): cv.string,
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the CO2signal sensor."""
+    token = config[CONF_TOKEN]
+    lat = config.get(CONF_LATITUDE, hass.config.latitude)
+    lon = config.get(CONF_LONGITUDE, hass.config.longitude)
+    country_code = config.get(CONF_COUNTRY_CODE)
+
+    _LOGGER.debug("Setting up the sensor using the %s", country_code)
+
+    devs = []
+
+    devs.append(CO2Sensor(token,
+                          country_code,
+                          lat,
+                          lon))
+    add_entities(devs, True)
+
+
+class CO2Sensor(Entity):
+    """Implementation of the CO2Signal sensor."""
+
+    def __init__(self, token, country_code, lat, lon):
+        """Initialize the sensor."""
+        self._token = token
+        self._country_code = country_code
+        self._latitude = lat
+        self._longitude = lon
+
+        if country_code is not None:
+            self._device_name = country_code
```

Comment:
`_device_name` is only used locally here, so it doesn't need to be an instance attribute. Just make it a local variable.